### DISCUSSION TITLE
Add Microsoft provider

### DISCUSSION THREAD

**Discussion by MicheleIRDG** [kreait/firebase-php#945] [Q&A]
### Describe the feature you would like to see

Hi
I have a firebase user created with mail
i need to allow that user ti log-in with microsoft...

is there a function for link new provider to the user?
i have see that "$signInResult = $auth->signInWithIdpAccessToken($provider, $accessToken, $redirectUrl = null, $oauthTokenSecret = null, $linkingIdToken);" but i don't understand how use it in this casecan i have a little demo?

**Comment by jeromegamez** [comment]
I haven't done this myself, but the provider should be `microsoft.com`, the access token is the access token you get from Microsoft after signing in with your Microsoft account. The ID token is from the Firebase User you want to link to the Microsoft account.

https://firebase.google.com/docs/projects/provisioning/configure-oauth#add-idp

As an alternative, you can (perhaps) also OAuth credentials of the Microsoft account.

The SDK uses the Firebase Authentication REST API, and its documentation is a little sparse, but perhaps you can work something out from it.

https://firebase.google.com/docs/reference/rest/auth#section-sign-in-with-oauth-credential

Again, I myself haven't used this before, so I'm afraid I can't be of much help, and I don't have the resources/time to set something up myself for free, but feel free to share your findings and further questions - perhaps I can contribute something or someone else can jump in.

(Please consider rewriting your original post title and contents, I migrated your feature request to a discussion)
